pkg/config/config_linux.go

```diff
@@ -24,3 +33,115 @@ func (c *RuntimeConfig) ValidatePinnsPath(executable string) error {
 
 	return err
 }
+
+// checkKernelRROMountSupport checks the kernel support for the Recursive Read-only (RRO) mounts.
+func checkKernelRROMountSupport() error {
+	kernelRROSupportOnce.Do(func() {
+		kernelRROSupportError = (func() error {
+			// Check the current kernel version for RRO mounts support...
+			err := validateKernelRROVersion()
+			if err != nil {
+				versionErr := err
+				// ... and if the kernel version does not match the minimum required version,
+				// then verify whether the kernel supports RRO mounts regardless, as often
+				// Linux distributions provide heavily patched kernel releases, and the
+				// current kernel might include backported support.
+				err = validateKernelRROMount()
+				if err != nil {
+					err = fmt.Errorf("%w: %w", versionErr, err)
+				}
+			}
+			return err
+		})()
+	})
+
+	return kernelRROSupportError
+}
+
+// validateKernelRROVersion checks whether the current kernel version matches the release 5.12 or newer,
+// which is the minimum required kernel version that supports Recursive Read-only (RRO) mounts.
+func validateKernelRROVersion() error {
+	const kernelVersion = 512
+
+	var (
+		uname        unix.Utsname
+		major, minor int
+	)
+
+	if err := unix.Uname(&uname); err != nil {
+		return fmt.Errorf("unable to retrieve kernel version: %w", err)
+	}
+
+	index := bytes.IndexByte(uname.Release[:], 0)
+	version := string(uname.Release[:index])
+
+	_, err := fmt.Sscanf(version, "%d.%d", &major, &minor)
+	if err != nil {
+		return fmt.Errorf("unable to parse kernel version %q: %w", version, err)
+	}
+
+	supportedMajor := major > kernelVersion/100
+	supportedMinor := major == kernelVersion/100 && minor >= kernelVersion%100
+
+	if !supportedMajor && !supportedMinor {
+		return fmt.Errorf("kernel version %q does not support recursive read-only mounts", version)
+	}
+
+	return nil
+}
```

Comment:
Maybe it's easier to use github.com/containers/storage/pkg/parsers/kernel rather than reimplement uname parsing here?

Reply:
I had looked at this package before; it was quite elaborate, given what we needed here. However, I will switch to it to benefit from a much more widely used and better-tested implementation. 👍

Reply:
Done.